src/laser_geometry.cpp

```diff
@@ -107,7 +117,7 @@ namespace laser_geometry
     for (unsigned int index = 0; index< scan_in.ranges.size(); index++)
     {
       const float range = ranges(0, index);
-      if (preservative || ((range < range_cutoff) && (range >= scan_in.range_min))) //if valid or preservative
+      if ((preservative && std::isfinite(ranges(0,index))) || ((range < range_cutoff) && (range >= scan_in.range_min))) //if valid or preservative
```

Comment:
If you handle of the non-finite cases before, you don't need to do this. Maybe only for `NaN` if it doesn't get handled.